flamingpy/simulations.py

```diff
@@ -286,25 +242,16 @@ def run_ec_simulation(
     else:
         raise ValueError(f"Decoder {params['decoder']} is either invalid or not yet implemented.")
 
-    # The Monte Carlo simulations
-    code = SurfaceCode
-    code_args = {key: params[key] for key in ["distance", "ec", "boundaries"]}
-
-    noise = noise_dict[params["noise"]]
-    if params.get("noise") == "iid":
-        if params.get("error_probability") is None:
-            raise ValueError("No argument `err_prob` found for `iid` noise.")
-        noise_args = {"error_probability": params.get("error_probability")}
-    else:
-        noise_args = {key: params[key] for key in ["delta", "p_swap"]}
+    noise = str_dict[params["noise"]]
+    code = str_dict[params["code"]]
 
-    decoder = params["decoder"]
     args = {
         "trials": params["trials"],
         "code": code,
```

Comment:
```suggestion
        "code": eval(params["code"]),
```

Reply:
Using `eval()` is a big no-no in Python except on very rare occasions, very unsafe and exposes you to many new issues. 

I have implemented an equivalent safe solution to your suggestion above.